clang/include/clang/AST/ASTNodeTraverser.h

```diff
@@ -458,6 +532,9 @@ class ASTNodeTraverser
     if (Traversal == TK_IgnoreUnlessSpelledInSource && D->isCXXForRangeDecl())
       return;
 
+    if (VisitLocs)
```

Comment:
Is that the only place where `VisitLocs` is used?

Reply:
Yes, there may be others but for now this is the only place I found where a Loc node contains a non-Loc node which contains a Loc node (and so has to decide whether to dump it).